config/snapshot/config_snapshot_test.go

```diff
@@ -108,6 +108,96 @@ func TestObserve_OnInit(t *testing.T) {
 	}
 }
 
+func TestObserve_OnInitRetry(t *testing.T) {
+	tests := []struct {
+		name             string
+		config           *configpb.Configuration
+		configGeneration int64
+		openErr          error
+		openOnRetry      bool
+		expectInitial    *configpb.Dashboard
+		expectError      bool
+	}{
+		{
+			name: "Reads config on retry",
+			config: &configpb.Configuration{
+				Dashboards: []*configpb.Dashboard{
+					{
+						Name: "dashboard",
+					},
+				},
+			},
+			configGeneration: 1,
+			expectInitial: &configpb.Dashboard{
+				Name: "dashboard",
+			},
+			openErr:     errors.New("fake error"),
+			openOnRetry: true,
+		},
+		{
+			name: "Returns error if config isn't present on retry",
+			config: &configpb.Configuration{
+				Dashboards: []*configpb.Dashboard{
+					{
+						Name: "dashboard",
+					},
+				},
+			},
+			configGeneration: 1,
+			expectInitial: &configpb.Dashboard{
```

Comment:
Do we expect an initial config in this test case? It looks like we'd expect just an error. "openOnRetry" is false, so it's just going to keep failing, like if the file just isn't there.

This _happens_ to succeed because your test code returns early after "expectError" and never actually checks this. It's fine to return early so we don't wait 30 seconds for a config we don't expect, but please remove it from the test setup array here.